src/Mix.php

```diff
@@ -74,13 +83,46 @@ public function __invoke($path, $manifestDirectory = ''): string
         }
 
         $manifest = self::$manifests[$manifestPath];
-        if (! isset($manifest[$path])) {
+
+        $path = self::resolvePath($path);
+
+        if (!isset($manifest[$path])) {
             throw new Exception("Unable to locate AssetMix file: {$path}.");
         }
 
         return $urlDomain . $manifestDirectory . $manifest[$path];
     }
 
+    /**
+     * This will strip the subdir part off of /subdir/js/app.js?1682657485
+     * 
+     * If asset timestamping is enabled then it will remove the timestamp 
+     * query string
+     * 
+     * '/subdir/js/app.js?1682657485'
+     * 
+     * will become
+     * '/js/app.js' 
+     * 
+     * @param mixed $path 
+     * @return mixed 
+     */
+    private static function resolvePath($path)
+    {
+        $pattern = sprintf(
+            '@^%s@',
+            self::$request->getAttribute('base')
+        );
+
+        // replace the /subdir if it exists at the beginning of the string
+        $path = preg_replace($pattern, '', $path);
+
+        // strip the asset timestamp query string
+        $path = explode('?', $path)[0];
```

Comment:
To strip timestamp from query string, there is already a PR that got merged in `cake3` branch. I suggest, you replicate that approach here too.

Ref. https://github.com/ishanvyas22/asset-mix/pull/40/

Reply:
Yes much less verbose.